hamocc/mo_parambgc_ini.F90

```diff
@@ -0,0 +1,1241 @@
+    call readjust_param()          ! potentially readjust namlist parameter-dependent parameters
+    call rates_2_timestep()        ! Converting rates from /d... to /dtb
+
+    call write_parambgc()          ! write out used parameters and calculate back rates from /dtb to /d..
+  end subroutine
+
+
+  !---------------------------------------------------------------------------------------------------------------------------------
+  subroutine ini_param_atm()
+    !
+    ! Atmospheric concentrations (atm_co2 is set via BGCNML namelist).
+    !
+    atm_o2      = 196800.
+    atm_n2      = 802000.
+
+#ifdef natDIC
+    atm_co2_nat = 284.32 ! CMIP6 pre-industrial reference
+#endif
+#ifdef BROMO
+    !For now use 3.4ppt from Hense and Quack (2009; Biogeosciences) NEED TO
+    !BE UPDATED WITH Ziska et al. (2013) climatology database
+    atm_bromo   = 3.4
+#endif
+#ifdef cisonew
+    ! set standard carbon isotope ratios
+    re1312      = 0.0112372
+    re14to      = 1.170e-12  ! Karlen et al. 1965 / Orr et al. 2017
+    ! set preindustr. d13c and bigd14C in atmosphere
+    prei13      = -6.5
+    prei14      =  0.
+#endif cisonew
+  end subroutine
+  
+
+  !---------------------------------------------------------------------------------------------------------------------------------
+  subroutine calc_param_atm()
+    !
+    ! AFTER having read the namelist:
+    ! calculate parameters for atmosphere from given parameters 
+    !
+#ifdef cisonew
+    beta13   = (prei13/1000.)+1.
+    alpha14  = 2.*(prei13+25.)
+    d14cat   = (prei14+alpha14)/(1.-alpha14/1000.)
+    ! calculate atm_c13 and atm_c14
+    atm_c13  = beta13*re1312*atm_co2/(1.+beta13*re1312)
+    d13C_atm = (((atm_c13/(atm_co2-atm_c13))/re1312)-1.)*1000.
+    ! absolute 14c concentration in preindustrial atmosphere
+    atm_c14  = ((d14cat/1000.)+1.)*re14to*atm_co2
+    ! factor for normalizing 14C tracers (~1e-12)
+    c14fac   = atm_c14/atm_co2
+#endif
+  end subroutine 
+
+  !---------------------------------------------------------------------------------------------------------------------------------
+  subroutine ini_fields_atm(kpie,kpje)
+    ! AFTER having read the nml: 
+    ! Initialise atmosphere fields. We use a 2D representation of atmospheric
+    ! fields for simplicity, even for cases where actually only a scalar value 
+    ! is used. The overhead of this is small. If an atm-field is present in
+    ! restart file (if BOXATM is activated), this will be overwritten later.
+
+    INTEGER, intent(in) :: kpie,kpje
+
+    ! local variables
+    INTEGER :: i,j
+  
+    DO j=1,kpje
+    DO i=1,kpie 
+      atm(i,j,iatmco2)  = atm_co2
+      atm(i,j,iatmo2)   = atm_o2
+      atm(i,j,iatmn2)   = atm_n2
+#ifdef natDIC
+      atm(i,j,iatmnco2) = atm_co2_nat
+#endif   
+#ifdef cisonew
+      atm(i,j,iatmc13)  = atm_c13
+      atm(i,j,iatmc14)  = atm_c14/c14fac
+#endif
+#ifdef BROMO
+      atm(i,j,iatmbromo)= atm_bromo
+#endif
+    ENDDO
+    ENDDO
+  end subroutine
+
```

Comment:
These 3 subroutines above could be simplified and combined into one routine if we would follow the idea of defining all parameters inside this module (as opposed to importing them)

Reply:
Hi @JorgSchwinger , I -partially- agree - the calculation and field initialization need to be carried out after reading the namelist, while the initial default initialization could be done before.

Reply:
I would leave it for now and re-think about it, once introducing the `protect` statements after #264 was coming in.